src/bsoncxx/test/bson_value.cpp

```diff
@@ -186,8 +186,8 @@ TEST_CASE("types::bson_value::value", "[bsoncxx::types::bson_value::value]") {
         bson_value::value value = elem.get_owning_value();
 
         SECTION("can create new views") {
-            bson_value::view view{value};
-            bson_value::view new_view{value};
+            bson_value::view view{static_cast<bson_value::view>(value)};
+            bson_value::view new_view{static_cast<bson_value::view>(value)};
```

Comment:
Fixes this warning on clang 3.6.2. I do not know why older clang is unable to infer the cast, since bson_value::value has a `operator bson_value::view() const noexcept;` definition.
```
[ 12%] Building CXX object src/bsoncxx/test/CMakeFiles/test_bson.dir/bson_value.cpp.o
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/test/bson_value.cpp:189:30: error: 
      no matching constructor for initialization of 'bson_value::view'
            bson_value::view view{value};
                             ^   ~~~~~~~
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:262:5: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'void *' for 1st argument; take the address of the
      argument with &
    view(void* internal_value);
    ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:19:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_double' for 1st
      argument
BSONCXX_ENUM(double, 0x01)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:20:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_utf8' for 1st argument
BSONCXX_ENUM(utf8, 0x02)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:21:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_document' for 1st
      argument
BSONCXX_ENUM(document, 0x03)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:22:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_array' for 1st argument
BSONCXX_ENUM(array, 0x04)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:23:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_binary' for 1st
      argument
BSONCXX_ENUM(binary, 0x05)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:24:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_undefined' for 1st
      argument
BSONCXX_ENUM(undefined, 0x06)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:25:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_oid' for 1st argument
BSONCXX_ENUM(oid, 0x07)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:26:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_bool' for 1st argument
BSONCXX_ENUM(bool, 0x08)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:27:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_date' for 1st argument
BSONCXX_ENUM(date, 0x09)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:28:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_null' for 1st argument
BSONCXX_ENUM(null, 0x0A)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:29:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_regex' for 1st argument
BSONCXX_ENUM(regex, 0x0B)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:30:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_dbpointer' for 1st
      argument
BSONCXX_ENUM(dbpointer, 0x0C)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:31:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_code' for 1st argument
BSONCXX_ENUM(code, 0x0D)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:32:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_symbol' for 1st
      argument
BSONCXX_ENUM(symbol, 0x0E)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:33:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_codewscope' for 1st
      argument
BSONCXX_ENUM(codewscope, 0x0F)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:34:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_int32' for 1st argument
BSONCXX_ENUM(int32, 0x10)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:35:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_timestamp' for 1st
      argument
BSONCXX_ENUM(timestamp, 0x11)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:36:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_int64' for 1st argument
BSONCXX_ENUM(int64, 0x12)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:37:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_decimal128' for 1st
      argument
BSONCXX_ENUM(decimal128, 0x13)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:38:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_maxkey' for 1st
      argument
BSONCXX_ENUM(maxkey, 0x7F)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:39:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_minkey' for 1st
      argument
BSONCXX_ENUM(minkey, 0xFF)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:64:5: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'const bsoncxx::v_noabi::types::bson_value::view &'
      for 1st argument
    view(const view&) noexcept;
    ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:261:5: note: 
      candidate constructor not viable: requires 4 arguments, but 1 was provided
    view(const std::uint8_t* raw, std::uint32_t length, std::uint32_t of...
    ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/test/bson_value.cpp:190:30: error: 
      no matching constructor for initialization of 'bson_value::view'
            bson_value::view new_view{value};
                             ^       ~~~~~~~
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:262:5: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'void *' for 1st argument; take the address of the
      argument with &
    view(void* internal_value);
    ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:19:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_double' for 1st
      argument
BSONCXX_ENUM(double, 0x01)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:20:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_utf8' for 1st argument
BSONCXX_ENUM(utf8, 0x02)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:21:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_document' for 1st
      argument
BSONCXX_ENUM(document, 0x03)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:22:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_array' for 1st argument
BSONCXX_ENUM(array, 0x04)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:23:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_binary' for 1st
      argument
BSONCXX_ENUM(binary, 0x05)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:24:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_undefined' for 1st
      argument
BSONCXX_ENUM(undefined, 0x06)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:25:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_oid' for 1st argument
BSONCXX_ENUM(oid, 0x07)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:26:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_bool' for 1st argument
BSONCXX_ENUM(bool, 0x08)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:27:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_date' for 1st argument
BSONCXX_ENUM(date, 0x09)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:28:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_null' for 1st argument
BSONCXX_ENUM(null, 0x0A)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:29:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_regex' for 1st argument
BSONCXX_ENUM(regex, 0x0B)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:30:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_dbpointer' for 1st
      argument
BSONCXX_ENUM(dbpointer, 0x0C)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:31:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_code' for 1st argument
BSONCXX_ENUM(code, 0x0D)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:32:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_symbol' for 1st
      argument
BSONCXX_ENUM(symbol, 0x0E)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:33:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_codewscope' for 1st
      argument
BSONCXX_ENUM(codewscope, 0x0F)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:34:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_int32' for 1st argument
BSONCXX_ENUM(int32, 0x10)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:35:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_timestamp' for 1st
      argument
BSONCXX_ENUM(timestamp, 0x11)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:36:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_int64' for 1st argument
BSONCXX_ENUM(int64, 0x12)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:37:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_decimal128' for 1st
      argument
BSONCXX_ENUM(decimal128, 0x13)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:38:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_maxkey' for 1st
      argument
BSONCXX_ENUM(maxkey, 0x7F)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/enums/type.hpp:39:1: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'bsoncxx::v_noabi::types::b_minkey' for 1st
      argument
BSONCXX_ENUM(minkey, 0xFF)
^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:60:42: note: 
      expanded from macro 'BSONCXX_ENUM'
#define BSONCXX_ENUM(type, val) explicit view(b_##type) noexcept;
                                         ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:64:5: note: 
      candidate constructor not viable: no known conversion from
      'bson_value::value' to 'const bsoncxx::v_noabi::types::bson_value::view &'
      for 1st argument
    view(const view&) noexcept;
    ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/types/bson_value/view.hpp:261:5: note: 
      candidate constructor not viable: requires 4 arguments, but 1 was provided
    view(const std::uint8_t* raw, std::uint32_t length, std::uint32_t of...
    ^
```

Reply:
Hm.  I really don't like that we need this cast here.  A more palatable way to do it would be to call the view() operator on value `bson_value::view view{value.view()};` but that shouldn't be necessary, either.  I would love to get to the bottom of why this is broken on travis.

I read a bug report about list initialization on some versions of clang ignoring implicit conversions.  Could you try changing this to use () instead of {} initialization?

`bson_value::view view(value);`

Normally, I prefer to call constructors with {} but let's see if using parentheses instead fixes this one.

Reply:
Awesome, glad that worked!